src/worldgen/biome/Climate.ts

```diff
@@ -260,14 +263,16 @@ export namespace Climate {
 			return space
 		}
 
-		search(values: number[], distance: DistanceMetric<T>): RLeaf<T> {
-			let dist = Infinity
-			let leaf: RLeaf<T> | null = null
+		search(values: number[], closest_leaf: RLeaf<T> | null, distance: DistanceMetric<T>): RLeaf<T> {
+			let dist = closest_leaf ? distance(closest_leaf, values) : Infinity
```

Comment:
To benchmark without optimization, change this line to say `let dist = Infinity`